src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2481,12 +2481,34 @@ namespace rpc {
     };
   };
 
+  // Resolves a provided address, will return the address if valid or the the resolved ONS wallet mapping if valid ONS Name
+  struct ONS_RESOLVE_ADDRESS : PUBLIC
```

Comment:
I don't think this RPC endpoint should be `PUBLIC`: we don't ever want public RPC nodes to be able to learn the decryption keys but if this if public it might induce someone to do all the lookups through public nodes which we don't want.

```suggestion
  struct ONS_RESOLVE_ADDRESS : RPC_COMMAND
```
will make it admin-only, which will still allow someone running a local oxend to use it, but won't make it available via public RPC nodes.

Reply:
Actually, thinking about this some more, is this endpoint really needed at all?  It seems as though the wallet could go through the `"ons_resolve"` endpoint instead, get back the encrypted value, then decrypt it locally: that way oxend *never* has the decryption key, but rather it just stays local to the wallet (or whatever client is accessing it).